docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,18 @@
+# Charm architecture
+
+A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators) is a set of Python scripts and libraries that operates some software. This charm operates a set of [GitHub self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) while handling the security and hardware management.
+
+Conceptually, the charm can be divided into the following:
+
+- Management of LXD ephemeral virtual machines to host [ephemeral self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/autoscaling-with-self-hosted-runners#using-ephemeral-runners-for-autoscaling).
+- Management of a [Python web service for checking GitHub repository settings](https://github.com/canonical/repo-policy-compliance).
+
+## LXD ephemeral virtual machines
+
+Using LXD ephemeral virtual machines to host the self-hosted runners ensures the environment is clean and isolated. The charm spawns a set of virtual machines with resources configured according to the charm configurations. The self-hosted runners are started with the ephemeral option and would clean up itself once it has taken a job from GitHub. At the end of the execution of the self-hosted runners application, the LXD ephemeral virtual machines would stop itself and clean up its resources. Freeing the resources to be used by future instances of virtual machines.
```

Comment:
s/Using LXD ephemeral virtual machines to host the self-hosted runners ensures the environment is clean and isolated. The charm spawns a set of virtual machines with resources configured according to the charm configurations./ To ensure a clean and isolated environment for every runner, Self-hosted runners are hosted using LXD virtual machines. The charm spawns virtual machines setting resources based on charm configurations.